gateway/WolkGateway.h

```diff
@@ -147,9 +148,9 @@ class WolkGateway : public connect::WolkSingle
 
     std::atomic<bool> m_localConnected;
 
-    std::shared_ptr<DeviceRepository> m_cacheDeviceRepository;
+    std::shared_ptr<InMemoryDeviceRepository> m_cacheDeviceRepository;
```

Comment:
So that we could invoke a specific method `loadInformationFromPersistentRepository()` in `WolkGateway.cpp:139`.

Reply:
But what is then the purpose of interface? Can't we have some load method in interface? And why do we event have in memory implementation if it in fact loads stuff from disk?

Reply:
The `RegistrationService` that also serves as a `DeviceFilter` will use just one pointer to a `DeviceRepository`. So it will use just whichever repository has been given to it. Only `WolkGateway` will treat the `cache` specially by taking the data from the persistent storage and loading it, and making it available in memory - which is the optimal way of operating for a `DeviceFilter` that should provide information whether device information exists or no. It's much better to do it from memory, rather than executing a SQL query on a file-system based database. Regular persistence is just used for what it says, persisting the data, just so we always have a copy of the data. Because of that, we don't need to obtain the entire list of devices the next time, just the ones that have been added after we last checked. And in case something fails, we also still have some thing to work off. There is a `DeviceStoragePolicy` enum in the `WolkGatewayBuilder.h` that also allows this to be fully configurable to the user, so that the user can choose whatever they would like to use.